packages/diagrams/lib/src/application_lifecycle.dart

```diff
@@ -0,0 +1,318 @@
+    required this.fillColor,
+    this.orientation = AxisDirection.right,
+    this.length = 400,
+    this.tipSize = 20,
+    this.thickness = 2,
+    required this.label,
+  });
+
+  final Color fillColor;
+  final AxisDirection orientation;
+  final Widget label;
+  final double thickness;
+  final double tipSize;
+  final double length;
+
+  @override
+  Widget build(BuildContext context) {
+    int quarterTurns = 0;
+    bool isVertical;
+    switch (orientation) {
+      case AxisDirection.up:
+        isVertical = true;
+        quarterTurns = 3;
+        break;
+      case AxisDirection.right:
+        isVertical = false;
+        quarterTurns = 0;
+        break;
+      case AxisDirection.down:
+        isVertical = true;
+        quarterTurns = 1;
+        break;
+      case AxisDirection.left:
+        isVertical = false;
+        quarterTurns = 2;
+        break;
+    }
+
+    final Widget arrow = RotatedBox(
+      quarterTurns: quarterTurns,
+      child: CustomPaint(
+        painter: ArrowPainter(fillColor: fillColor, thickness: thickness),
+        child: SizedBox(
+          height: tipSize,
+          width: length,
+        ),
+      ),
+    );
+
+    if (isVertical) {
+      return Row(
+        children: <Widget>[
+          if (orientation == AxisDirection.down) label,
+          arrow,
+          if (orientation == AxisDirection.up) label,
+        ],
+      );
+    } else {
+      return Column(
+        children: <Widget>[
+          if (orientation == AxisDirection.right) label,
+          arrow,
+          if (orientation == AxisDirection.left) label,
+        ],
+      );
+    }
+  }
+}
+
+class ArrowPainter extends CustomPainter {
+  const ArrowPainter({required this.fillColor, this.thickness = 10, this.orientation = AxisDirection.right});
+
+  final Color fillColor;
+  final AxisDirection orientation;
+  final double thickness;
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    final Offset center = size.center(Offset.zero);
+    final List<Offset> points = <Offset>[
+      Offset(0, center.dy - thickness),
+      Offset(size.width - size.height, center.dy - thickness),
+      Offset(size.width - size.height, 0),
+      Offset(size.width, center.dy),
+      Offset(size.width - size.height, size.height),
+      Offset(size.width - size.height, center.dy + thickness),
+      Offset(0, center.dy + thickness),
+    ];
+
+    final Path arrowPath = Path()..addPolygon(points, true);
+
+    canvas.drawPath(
+      arrowPath,
+      Paint()..color = fillColor,
+    );
+
+    canvas.drawPath(
+      arrowPath,
+      Paint()
+        ..style = PaintingStyle.stroke
+        ..strokeWidth = 1.0
+        ..color = AppLifecycleDiagram.arrowBorderColor,
+    );
+  }
+
+  @override
+  bool shouldRepaint(CustomPainter oldDelegate) => false;
+}
+
+class AppLifecycleDiagramStep extends DiagramStep {
+  @override
+  final String category = 'dart-ui';
+
+  @override
+  Future<List<AppLifecycleDiagram>> get diagrams async =>
```

Comment:
nit: in flutter style this should be in {} instead of => since it doesn't fit on one line.

Reply:
Done.